validator/server_api/json.go

```diff
@@ -99,19 +95,23 @@ func ValidationInputToJson(entry *validator.ValidationInput) *InputJSON {
 		DelayedMsgB64: base64.StdEncoding.EncodeToString(entry.DelayedMsg),
 		StartState:    entry.StartState,
 		PreimagesB64:  jsonPreimagesMap,
-		UserWasms:     make(map[common.Hash]UserWasmJson),
+		UserWasms:     make(map[string]map[common.Hash]string),
 		DebugChain:    entry.DebugChain,
 	}
 	for _, binfo := range entry.BatchInfo {
 		encData := base64.StdEncoding.EncodeToString(binfo.Data)
 		res.BatchInfo = append(res.BatchInfo, BatchInfoJson{Number: binfo.Number, DataB64: encData})
 	}
-	for moduleHash, info := range entry.UserWasms {
-		encWasm := UserWasmJson{
-			Asm:    base64.StdEncoding.EncodeToString(info.Asm),
-			Module: base64.StdEncoding.EncodeToString(info.Module),
+	for arch, wasms := range entry.UserWasms {
+		archWasms := make(map[common.Hash]string)
+		for moduleHash, data := range wasms {
+			compressed, err := arbcompress.CompressLevel(data, 1)
+			if err != nil {
+				continue
+			}
+			archWasms[moduleHash] = base64.StdEncoding.EncodeToString(compressed)
```

Comment:
At some point it might be worth doing something like `*jsonapi.PreimagesMapJson` to optimize this and avoid extra copying but that probably belongs in a separate PR. As-is it's still more efficient than what we previously had.